app/components/Texting/Texting.jsx

```diff
@@ -0,0 +1,223 @@
+        </h1>
+        <h3 className="description">You will receive their address and phone number.</h3>
+      </div>
+
+      <div className="inputFields">
+        <div className="inputField">
+          <div className="label">First name (optional)</div>
+          <input
+            type="text"
+            name="userName"
+            className="input"
+            value={userName}
+            data-field="name"
+            onChange={onChange}
+          />
+        </div>
+
+        <div className="inputField">
+          <div className="label">Phone number (required)</div>
+          <div className="dataRates">*Standard text and data rates may apply.</div>
+          <input
+            type="text"
+            name="phoneNumber"
+            className="input"
+            value={phoneNumber}
+            data-field="phoneNumber"
+            onChange={onChange}
+          />
+        </div>
+
+        <label className="checkBox dataRates">
+          I agree to receive text messages from SF Service Guide.
+          <input
+            type="checkbox"
+            name="agreed"
+            checked={agreed}
+            data-field="agree"
+            onChange={onChange}
+          />
+        </label>
+      </div>
+
+      <div className="buttons">
+        <button
+          type="button"
+          className="cancel button"
+          onClick={toggle}
+        >
+          cancel
+        </button>
+        <button
+          type="submit"
+          className={`button${agreed && phoneNumberValidator(phoneNumber) ? '' : ' disabled'}`}
+          disabled={!(agreed && phoneNumberValidator(phoneNumber))}
+          data-field="submit"
+          onClick={onSubmit}
+        >
+          Text me
+        </button>
+      </div>
+
+      <div className="privacy">
+        We will not share or sell your information with third party sites.
+        For more information on privacy details, please go
+        <a target="_blank" rel="noopener noreferrer" href="https://sfserviceguide.org/"> here </a>
+        to view.
+      </div>
+    </>
+  );
+
+  // Loading spinner: Show a loading spinner when the request being processed in the Backend
+  const loadingView = () => (<Loader />);
+
+  // Success sending view
+  const sentView = () => (
+    <div className="responseIcons">
+      <div className="circle">
+        <img src={images.icon('check-mark')} alt="check-mark" className="responseIcon" />
+        <h1 className="responseText" data-field="sent">Sent!</h1>
+      </div>
+    </div>
+  );
+
+  // Error View: show Alert graphic and a friendly message
+  const errorView = () => (
+    <div className="responseIcons error">
+      <img src={images.icon('emergency')} alt="error" className="responseIcon" />
+      <h1 className="responseText error">Sorry, something went wrong, please try again later.</h1>
+    </div>
+  );
+
+  const view = () => {
+    if (error) {
+      return errorView();
```

Comment:
I like how you've broken these out! I don't think you need to wrap them in functions, though. I think you should be able to get away with just assigning them directly to the variables like:

```
const loadingView = <Loader />;

...

const view = () => {
    ... 
    if(isLoading) {
        return loadingView;
    }
};
```

Reply:
Thank you @AnnaKalkanis 
I think we could just return Loader component like this:

 ```
 const view = () => {
  . . .
    if (isLoading) {
      return <Loader />;
    }
  . . .
  };
```